src/pyhf/utils.py

```diff
@@ -67,3 +68,35 @@ def convert(self, value, param, ctx):
             self.fail(
                 '{0:s} is not a valid equal-delimited string'.format(value), param, ctx
             )
+
+
+def digest(obj, algorithm='sha256'):
+    """
+    Get the digest for the provided object. Note: object must be JSON-serializable.
+
+    The hashing algorithms supported are in :module:`hashlib`, part of Python's Standard Libraries.
+
+    Raises:
+        ValueError: If the object is not JSON-serializable or if the algorithm is not supported.
+
+    Args:
+        obj (`obj`): A JSON-serializable object to compute the digest of. Usually a :class:`~pyhf.workspace.Workspace` object.
+        algorithm (`str`): The hashing algorithm to use.
+
+    Returns:
+        digest (`str`): The digest for the JSON-serialized object provided and hash algorithm specified.
+    """
+
+    try:
+        stringified = json.dumps(obj, sort_keys=True, ensure_ascii=False).encode('utf8')
```

Comment:
Not super crazy about "stringified" as a name. Would prefer something along the lines of "json_as_str" but either will be fine.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

I like stringified :)